### PR TITLE
Update `compiler-builtins` and enable f128 tests on all non-buggy platforms

### PR DESCRIPTION
**Pull request by tgross35**
Update compiler_builtins to 0.1.138 and pin it. This updates to a new version of builtins that includes [1], which was
the last blocker to us enabling `f128` tests on all platforms.

With that, we now provide symbols necessary to work with `f128` everywhere. This means that we are no longer restricted to systems that provide `f128` symbols themselves, and can enable tests by default.

There are still a handful of platforms that need to remain disabled because of bugs and some that had to get updated.

Math support is still off by default since those symbols are not yet available.

[1]: https://github.com/rust-lang/compiler-builtins/pull/624

try-job: test-various
try-job: i686-gnu-nopt

### CODE REVIEW

**Comment by tgross35** [comment]
This includes https://github.com/rust-lang/rust/pull/132433 which includes https://github.com/rust-lang/rust/pull/132206, so the chain is a bit deep (I'm keeping updates separate from rust-lang/rust changes so we aren't blocked from updating further if something goes wrong here). But that doesn't mean we can't start some tests.

@bors try

**Comment by tgross35** [comment]
Those are quite a few try jobs but it still doesn't cover everything that will be newly enabled here.

@bors rollup=never

**Comment by tgross35** [comment]
Also cc @beetrees

**Comment by workingjubilee** [comment]
This has a merge conflict, it seems.

**Comment by workingjubilee** [comment]
I have rebased it.

**Comment by workingjubilee** [comment]
I do not believe we run tests on mips64

**Comment by workingjubilee** [comment]
I decided to change up the suite of tests slightly and rerun this. In particular, I noticed we didn't run the infamous `test-various`.

@bors try

**Comment by tgross35** [comment]
I updated this to skip x86 and mips as mentioned. Indeed we don't test this (and try job only allow the jobs listed [here](https://github.com/rust-lang-ci/rust/blob/b8c8287a229cd79604aa84c25e1235fc78cd5f2e/src/ci/github-actions/jobs.yml#L118), though I wish it was possible to do try jobs for tier 3 targets) but I know there are people running the test suite locally, e.g. https://github.com/rust-lang/rust/pull/127235.

Updated try jobs to those that have been mentioned but not yet tested.

@bors try

File: library/std/build.rs [hunk 1]
```diff
@@ -122,19 +122,24 @@ fn main() {
...
         // Unsupported <https://github.com/llvm/llvm-project/issues/94434>
         ("arm64ec", _) => false,
-        // ABI and precision bugs <https://github.com/rust-lang/rust/issues/125109>
-        // <https://github.com/rust-lang/rust/issues/125102>
-        ("powerpc" | "powerpc64", _) => false,
+        // Selection bug <https://github.com/llvm/llvm-project/issues/96432>
+        ("mips" | "mips64" | "mips32r6" | "mips64r6", _) => false,
         // Selection bug <https://github.com/llvm/llvm-project/issues/95471>
         ("nvptx64", _) => false,
+        // ABI and precision bugs <https://github.com/rust-lang/rust/issues/125109> et al. (full
```
[the remark this answers is not in the document]
Comment (tgross35) [reply library/std/build.rs]: Thanks, updated both of these

**Review by beetrees — APPROVED** [review]
LGTM. Thanks

**Comment by workingjubilee** [comment]
> All MIPS targets are tier 3, so no builds or tests of any kind are run by rust-lang/rust. The benefit of adding it to the match statement is just for those running tests locally/Linux distros with a MIPS64 port etc. 32-bit sparc is also tier 3 only and is already on the list.

mmh. my usual inclination is that we should leave known-failing-on-tier-3 tests on, so that people who are doing _just that_ are aware that _the platform's support is incomplete_... and if they run the tests and then are bothered by it, we're usually amenable to exclusions.

as this is a nightly feature though I'm not strongly motivated either way.

**Comment by workingjubilee** [comment]
let's get this show on the road:
@bors r+

**Comment by jethrogb** [comment]
Is it correct that compiler-builtins doesn't define `fmodl` (for f128)?

**Comment by jethrogb** [comment]
I see in your first message you mention:

> Math support is still off by default since those symbols are not yet available.

However, the compiler does appear to be generating references to `fmodl` with this PR merged.

**Pull request merged**